examples/platforms/nrf528xx/src/radio.c

```diff
@@ -701,22 +702,54 @@ otError otPlatRadioSetCcaEnergyDetectThreshold(otInstance *aInstance, int8_t aTh
     nrf_802154_cca_cfg_t ccaConfig;
 
     // The minimum value of ED threshold for radio driver is -94 dBm
-    if (aThreshold < NRF528XX_MIN_CCA_ED_THRESHOLD)
+    if (aThreshold + sLnaGain < NRF528XX_MIN_CCA_ED_THRESHOLD)
     {
         error = OT_ERROR_INVALID_ARGS;
     }
     else
     {
         memset(&ccaConfig, 0, sizeof(ccaConfig));
         ccaConfig.mode         = NRF_RADIO_CCA_MODE_ED;
-        ccaConfig.ed_threshold = nrf_802154_ccaedthres_from_dbm_calculate(aThreshold);
+        ccaConfig.ed_threshold = nrf_802154_ccaedthres_from_dbm_calculate(aThreshold + sLnaGain);
 
         nrf_802154_cca_cfg_set(&ccaConfig);
     }
 
     return error;
 }
 
+otError otPlatRadioGetFemLnaGain(otInstance *aInstance, int8_t *aGain)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    otError error = OT_ERROR_NONE;
+
+    if (aGain == NULL)
+    {
+        error = OT_ERROR_INVALID_ARGS;
+    }
+    else
+    {
+        *aGain = sLnaGain;
+    }
+
+    return error;
+}
+
+otError otPlatRadioSetFemLnaGain(otInstance *aInstance, int8_t aGain)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    int8_t threshold;
+
+    otEXPECT(otPlatRadioGetCcaEnergyDetectThreshold(aInstance, &threshold) == OT_ERROR_NONE);
+    otEXPECT(otPlatRadioSetCcaEnergyDetectThreshold(aInstance, threshold) == OT_ERROR_NONE);
+    sLnaGain = aGain;
```

Comment:
It's unclear to me here. Should `sLnaGain = aGain` come before `otEXPECT(otPlatRadioSetCcaEnergyDetectThreshold(aInstance, threshold) == OT_ERROR_NONE);` so that the set method could check the new `sLnaGain`?

Reply:
You are right. Made a mistake here.